### DISCUSSION TITLE
`x-on.prevent:mouseenter` broken after update to 3.14.0

### DISCUSSION THREAD

**Discussion by UlrichThomasGabor** [alpinejs/alpine#4268] [7. Bugs (archived)] [answered]
Something changed between [v3.13.10](https://github.com/alpinejs/alpine/releases/tag/v3.13.10) and [v3.14.0](https://github.com/alpinejs/alpine/releases/tag/v3.14.0) which causes an attribute like that:
```
x-on.prevent:mouseenter="console.log(1)"
```
to not work anymore.

**Comment by ekwoka** [comment, marked as answer]
Not really sure why the changes would impact this specific case, but you are supposed to put `:mouseenter` BEFORE `.prevent`

`x-on:mouseenter.prevent` [works fine](https://awesomealpine.com/play/?coreplugins=0&tw=1&html=PGRpdgogIGNsYXNzPSJmbGV4IGZsZXgtY29sIGdhcC00IGl0ZW1zLWNlbnRlciB3LWZ1bGwiCiAgeC1kYXRhPSJ7aG92ZXJlZDogJycgfSIKPgogIDxoMSBjbGFzcz0idGV4dC14bCI-CiAgICBIb3ZlcmluZzogPHNwYW4geC10ZXh0PSJob3ZlcmVkIj48L3NwYW4-CiAgICA8YnV0dG9uCiAgICAgIHR5cGU9ImJ1dHRvbiIKICAgICAgeC1vbi5wcmV2ZW50Om1vdXNlZW50ZXI9ImhvdmVyZWQgPSAkZWwudGV4dENvbnRlbnQiCiAgICA-CiAgICAgIGZpcnN0CiAgICA8L2J1dHRvbj4KICAgIDxidXR0b24KICAgICAgdHlwZT0iYnV0dG9uIgogICAgICB4LW9uOm1vdXNlZW50ZXIucHJldmVudD0iaG92ZXJlZCA9ICRlbC50ZXh0Q29udGVudCIKICAgID4KICAgICAgc2Vjb25kCiAgICA8L2J1dHRvbj4KICAgIDxkaXY-PC9kaXY-CiAgPC9oMT4KPC9kaXY-Cg&script=Y29uc29sZS5sb2coJ2hlbGxvJyk7Cg)

It is curious that the way you show worked before, but that's not an intended behavior as far as I'm aware

the relevant pr is #4209

**Reply by SimoTod** [reply]
I think that the modifier is coming through as `.prevent:mouseenter` (which was technically doing nothing before the change. I'm not sure there are any default actions to prevent anyway) and therefore isn't caught by https://github.com/alpinejs/alpine/blob/main/packages/alpinejs/src/utils/on.js#L115 (the previous version was exiting earlier so it was working by chance)

**Reply by ekwoka** [reply]
yeah, that would make sense.
